### PR TITLE
Add CsWinRT authoring samples

### PR DESCRIPTION
**Pull request by angelazhangmsft**
## Description

Adding C#/WinRT authoring samples for authoring custom WinUI controls. These samples are for 1.1.

Samples are waiting on bug: [task.ms/39285537](https://task.ms/39285537)

## Checklist

- [x] Samples build and run using the Visual Studio versions listed in the [Windows development docs](https://docs.microsoft.com/windows/apps/windows-app-sdk/set-up-your-development-environment?tabs=stable#2-install-visual-studio).
- [x] Samples build and run on all supported platforms (x64, x86, ARM64) and configurations (Debug, Release).
- [x] Samples set the minimum supported OS version to Windows 10 version 1809.
- [x] Samples build clean with no warnings or errors.
- [x] **[For new samples]**: Samples have completed the [sample guidelines checklist](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#checklist) and follow [standardization/naming guidelines](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#standardization-and-naming).
- [x] I have commented on my PR `/azp run` to have the CI build run on my branch. This must be done right before merging to ensure the build is up to date and accurate.

### CODE REVIEW

File: Samples/CustomControls/CsApp/CsApp.csproj [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    <TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
+    <RootNamespace>CsApp</RootNamespace>
+    <ApplicationManifest>app.manifest</ApplicationManifest>
+    <Platforms>x86;x64;arm64</Platforms>
+    <RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
+    <UseWinUI>true</UseWinUI>
+    <EnablePreviewMsixTooling>true</EnablePreviewMsixTooling>
+    <PublishProfile>Properties\PublishProfiles\win10-$(Platform).pubxml</PublishProfile>
+    <!-- TODO: remove workaround for below -->
+	<CsWinRTAuthoring_SkipDependencies>false</CsWinRTAuthoring_SkipDependencies>
```
Comment (angelazhangmsft) [inline Samples/CustomControls/CsApp/CsApp.csproj:14]: Remove before merging

File: Samples/CustomControls/WinUIComponentCs/WinUIComponentCs.csproj [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+		<TargetFramework>net6.0-windows10.0.19041.0</TargetFramework>
+		<TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
+		<RootNamespace>WinUIComponentCs</RootNamespace>
+		<RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
+		<UseWinUI>true</UseWinUI>
+		<CsWinRTComponent>true</CsWinRTComponent>
+		<WindowsAppContainer>true</WindowsAppContainer>
+		<Platforms>x64;x86;arm64</Platforms>
+		<!-- Workaround for .NET SDK update -->
+		<WindowsAppSDKVerifyWinrtRuntimeVersion>false</WindowsAppSDKVerifyWinrtRuntimeVersion>
```
Comment (angelazhangmsft) [inline Samples/CustomControls/WinUIComponentCs/WinUIComponentCs.csproj:12]: Remove before merging

File: Samples/CustomControls/WinUIComponentCs/WinUIComponentCs.csproj [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	</PropertyGroup>
+
+	<ItemGroup>
+		<PackageReference Include="Microsoft.Windows.CsWinRT" Version="1.5.0-prerelease.220214.2" />
```
Comment (angelazhangmsft) [inline Samples/CustomControls/WinUIComponentCs/WinUIComponentCs.csproj:16]: update NuGet paackages to public version before merging

File: Samples/CustomControls/WinUIComponentCs/WinUIComponentCs.csproj [hunk 4]
```diff
@@ -0,0 +1,27 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+	<PropertyGroup>
+		<TargetFramework>net6.0-windows10.0.19041.0</TargetFramework>
+		<TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
+		<RootNamespace>WinUIComponentCs</RootNamespace>
+		<RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
+		<UseWinUI>true</UseWinUI>
+		<CsWinRTComponent>true</CsWinRTComponent>
+		<WindowsAppContainer>true</WindowsAppContainer>
```
Comment (angelazhangmsft) [inline Samples/CustomControls/WinUIComponentCs/WinUIComponentCs.csproj:9]: confirm if this property needs to be documented.

File: Samples/CustomControls/CppApp/CppApp.vcxproj [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+      <Platform>x64</Platform>
+    </ProjectConfiguration>
+    <ProjectConfiguration Include="Release|arm64">
+      <Configuration>Release</Configuration>
+      <Platform>arm64</Platform>
+    </ProjectConfiguration>
+  </ItemGroup>
+  <PropertyGroup Label="Configuration">
+    <ConfigurationType>Application</ConfigurationType>
+    <PlatformToolset>v143</PlatformToolset>
```
Comment (erik0686) [inline Samples/CustomControls/CppApp/CppApp.vcxproj:58]: Please avoid defining the platform toolset here, please import this file instead:
https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/Samples/Build.Common.Cpp.props

For example: https://github.com/microsoft/WindowsAppSDK-Samples/blob/ffff65f950d4e3f098dd82be8df4d6898234b0aa/Samples/ResourceManagement/cpp-console-unpackaged/console_unpackaged_app.vcxproj#L6

Comment (angelazhangmsft) [reply Samples/CustomControls/CppApp/CppApp.vcxproj]: fixed!

**Review by erik0686 — APPROVED** [review]

**Comment by erik0686** [comment]
@angelazhangmsft have you rebased your branch on top of main lately? the CI build changed recently, so you might want to check how it is run now :)

It should be something like /azp run SamplesCI-<FeatureName>

**Pull request merged**